cmd/hub/mcscontroller/controller_test.go

```diff
@@ -0,0 +1,158 @@
+package mcscontroller_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	mcsv1a1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"github.com/submariner-io/lighthouse/cmd/hub/mcscontroller"
+	lhconst "github.com/submariner-io/lighthouse/pkg/constants"
+	"github.com/submariner-io/lighthouse/pkg/lhutil"
+	"github.com/submariner-io/lighthouse/pkg/mcs"
+)
+
+const (
+	serviceName = "svc"
+	serviceNS   = "svc-ns"
+	brokerNS    = "submariner-broker"
+	cluster1    = "c1"
+	cluster2    = "c2"
+)
+
+var (
+	service = &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      serviceName,
+			Namespace: serviceNS,
+		},
+	}
+	export = &mcsv1a1.ServiceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "placeholder",
+			Namespace: brokerNS,
+			Labels: map[string]string{
+				lhconst.LabelSourceNamespace:      serviceNS,
+				lhconst.LighthouseLabelSourceName: serviceName,
+			},
+		},
+	}
+)
+
+func TestImportGenerated(t *testing.T) {
```

Comment:
I guess we need more test cases, e.g conflict, update of an export, deletion of an export, ..

Reply:
correct. Tests listed in #27